In [1]:
import numpy as np
import pandas as pd

we queried for Guterres, António speeches and exported the result to csv \
find out which documents we need now

In [2]:
df_sg1 = pd.read_csv('test_data/search_guterres_1.csv')
df_sg2 = pd.read_csv('test_data/search_guterres_2.csv')
df_sg3 = pd.read_csv('test_data/search_guterres_3.csv')
df_sg = pd.concat([df_sg1, df_sg2, df_sg3])
print(df_sg.index.size)
# df_sg.head(3)

274


In [3]:
df_sg = df_sg[['Record Identifier', 'Related Item.Resource Type-1', 'Related Item.Edition-1', 'Publication Date', 'Last Modification Date', '989__a']]
df_sg.reset_index(inplace=True)
df_sg.head()

index  Record Identifier Related Item.Resource Type-1  \
0      0            4088514                    A/79/PV.7   
1      1            4072438                   A/78/PV.87   
2      2            4067194                 E/2024/SR.14   
3      3            4039683                    S/PV.9547   
4      4            4036553                 E/2023/SR.19   

  Related Item.Edition-1 Publication Date  Last Modification Date    989__a  
0                     A/              NaN            2.025092e+13  Speeches  
1                     A/              NaN            2.025012e+13  Speeches  
2                     E/              NaN            2.024113e+13  Speeches  
3                     S/              NaN            2.025091e+13  Speeches  
4                     E/              NaN            2.024081e+13  Speeches

In [4]:
# df_sg['searchable_document_name'] = [x.replace('/','\\/') for x in df_sg['Related Item.Resource Type-1']] # remove or escape '/' so it is searchable in the un library
df_sg['searchable_document_name'] = [x.replace('/','') for x in df_sg['Related Item.Resource Type-1']] # remove or escape '/' so it is searchable in the un library
print(f'there are {df_sg.searchable_document_name.nunique()} unique docs')

there are 238 unique docs


In [5]:
# print out these docs
print(f'All documents related to Guterres: {df_sg.searchable_document_name.tolist()}')

All documents related to Guterres: ['A79PV.7', 'A78PV.87', 'E2024SR.14', 'SPV.9547', 'E2023SR.19', 'SPV.9451', 'AC.577SR.5', 'A77PV.48', 'AES-11PV.17', 'E2022SR.25', 'SPV.9090', 'SPV.9064', 'SPV.9027', 'SPV.9001', 'SPV.8974', 'SPV.8906', 'SPV.8900', 'A75PV.59', 'A75PV.51', 'SPV.8803', 'AS-31PV.2', 'SPV.8721', 'SPV.8717', 'A73PV.74', 'A73PV.71', 'A73PV.59', 'A73PV.2', 'SPV.8185', 'A72PV.1', 'SPV.6949', 'E2011SR.39', 'SPV.9734', 'SPV.9662', 'SPV.9557', 'A77PV.74', 'SPV.9241', 'SPV.9220', 'SPV.9181', 'E2022SR.30', 'SPV.9135', 'A76PV.56', 'SPV.8886', 'A75PV.105', 'A75PV.67', 'SPV.8834', 'A75PV.72', 'A75PV.16', 'A74PV.66', 'A73PV.102', 'SPV.8626', 'A73PV.87', 'SPV.8548', 'SPV.8382', 'A72PV.83', 'SPV.8183', 'A72PV.80', 'E2017SR.43', 'SPV.8137', 'SPV.8080', 'A71PV.60', 'A71PV.27', 'SPV.7000', 'AC.367SR.32', 'SPV.6668', 'SPV.9939', 'SPV.9919', 'SPV.9884', 'A78PV.81', 'SPV.9731', 'AC.578SR.5', 'E2023SR.30', 'SPV.9346', 'SPV.9310', 'A76PV.75', 'A76PV.63', 'SPV.8967', 'A76PV.19', 'SPV.8923', 'SPV

In [6]:
df_sg.searchable_document_name.agg(['nunique','count','size'])
# seems like no None values

nunique    238
count      274
size       274
Name: searchable_document_name, dtype: int64

In [7]:
duplicat = df_sg.groupby(df_sg['searchable_document_name']).count()
duplicat[duplicat['index'] > 1].head(4)
# duplicat.columns

index  Record Identifier  \
searchable_document_name                             
A72PV.117                     2                  2   
A72PV.99                      2                  2   
A73PV.1                       2                  2   
A73PV.99                      2                  2   

                          Related Item.Resource Type-1  \
searchable_document_name                                 
A72PV.117                                            2   
A72PV.99                                             2   
A73PV.1                                              2   
A73PV.99                                             2   

                          Related Item.Edition-1  Publication Date  \
searchable_document_name                                             
A72PV.117                                      2                 2   
A72PV.99                                       2                 2   
A73PV.1                                        2                 2   
A73PV.99                                       2                 2   

                          Last Modification Date  989__a  
searchable_document_name                                  
A72PV.117                                      2       2  
A72PV.99                                       2       2  
A73PV.1                                        2       2  
A73PV.99                                       2       2

In [8]:
df_sg.groupby(df_sg['Related Item.Edition-1']).count()

index  Record Identifier  \
Related Item.Edition-1                             
A/                        112                112   
E/                         18                 18   
S/                        144                144   

                        Related Item.Resource Type-1  Publication Date  \
Related Item.Edition-1                                                   
A/                                               112                97   
E/                                                18                13   
S/                                               144               109   

                        Last Modification Date  989__a  \
Related Item.Edition-1                                   
A/                                         112     112   
E/                                          18      18   
S/                                         144     144   

                        searchable_document_name  
Related Item.Edition-1                            
A/                                           112  
E/                                            18  
S/                                           144

The documents are related to the following meetings:
- S: 144 docs - Security Council
- A: 112 docs - General Assembly 
- E: 018 docs - Economic and Social Council

In [15]:
df_sg[df_sg.searchable_document_name.str.contains('SPV.8789')]

Empty DataFrame
Columns: [index, Record Identifier, Related Item.Resource Type-1, Related Item.Edition-1, Publication Date, Last Modification Date, 989__a, searchable_document_name]
Index: []